src/main/java/io/harness/cf/client/api/InnerClient.java

```diff
@@ -335,7 +335,7 @@ public synchronized void waitForInitialization()
       throws InterruptedException, FeatureFlagInitializeException {
     while (!initialized) {
       log.info("Wait for initialization to finish");
-      wait();
+      wait(5000);
```

Comment:
Instead of `wait()` which waits forever (until a notify) the `wait(5000)` check back every 5 seconds. This behaves that same since it is in a loop, but allows the failure check to happen.